### PR TITLE
feat(helm): add topologySpreadConstraints value

### PR DESCRIPTION
**Pull request by ericpaulsen**
my attempt at closing #12904 - @spikecurtis let me know where i'm going wrong, the `topologySpreadConstraints` value is not included in the template despite being set in the `values.yaml`.

i'm running `helm template . --debug` to test.

### CODE REVIEW

File: helm/libcoder/templates/_coder.yaml [hunk 1]
```diff
@@ -43,6 +43,10 @@ spec:
       nodeSelector:
       {{ toYaml . | nindent 8 }}
       {{- end }}
+      {{- with .Values.coder.topologySpreadConstraints }}
+      topologySpreadConstraints:
+      {{- toYaml .Values.coder.topologySpreadConstraints | nindent 8 }}
+      {{- end }}
```
Comment (spikecurtis) [inline helm/libcoder/templates/_coder.yaml:49]: A `with X` block sets `.` equal to `X` within that block. 

c.f.

```
      {{- with .Values.coder.affinity }}
      affinity:
      {{- toYaml . | nindent 8 }}
      {{- end }}
```

It only refers to `.Values.coder.affinity` once, in the `with` clause. Then, we use `.` to refer to it when calling `toYaml`

Comment (ericpaulsen) [reply helm/libcoder/templates/_coder.yaml:49]: ah makes sense - although adding that does not fix the issue. i'll push a commit with this change for posterity.

Comment (spikecurtis) [reply helm/libcoder/templates/_coder.yaml:49]: Worked for me --- how are you testing?

**Review by johnstcn — COMMENTED** [review]
Tested it out and works on my own cluster after `helm dependency update --skip-refresh`

Just missing a doc string (and possibly a make gen)

**Review by spikecurtis — APPROVED** [review]

**Pull request merged**